File: web-dev/cs50-2021/C-Program/week1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
int LEN = 4;

int main(void)
{
 int coins[4] = {25, 10, 5, 1};
 float dollar;
 int count;
 int number = 0;

 //Get an input from user //Condition that must be fulfilled before program will continue
 do
 {
  dollar = get_float("Change: ");
 } while (dollar < 0);

 // Convert the dollar into coin value
 int change = round(dollar * 100);

 // Create a for-loop to calc the amount to be given without doing it one-by-one 
 for (int i = 0; i < LEN; i++)
 {
  if (coins[i] <= change)
  {
   // Loop through the coins as we reduce the amount to be given out
   count = change / coins[i];

   // Save the number of times a particualar coin has been used in the variable "number"
   number = count + number; 

   // Tracks the change to be given out to the customer
   change = change - count * coins[i];
  }
 }

 // Print out the coin number
 printf("%i\n", number);
 
}

File: web-dev/cs50-2021/C-Program/week6/cash/cash.py
import cs50  # importing the get_float function from cs50 libraries

# Defining our main function for the program


def main():
    # Setting the condition that the user must meet if not it will keep asking until right input is given
    while True:
        # Getting input from the user
        dollar = cs50.get_float("Change Owed: ")
    # The input should be a non-negative integer
        if dollar >= 0:
            break

    # Changing the dollar input into cents
    change = int(dollar * 100)

    # Calling the function to calculate the amount
    calc(change)

# defining our function to calculate the least number of coins to be given


def calc(n):
    number = 0
    coins = [25, 10, 5, 1]
    for coin in coins:
        if coin <= n:
            count = n // coin
            number += count
            n = n - count * coin
    print(f"{number}\n", end="")


# Call back main
if __name__ == '__main__':
    main()
